src/cargo/ops/vendor.rs

```diff
@@ -639,16 +639,13 @@ fn copy_and_checksum<T: Read>(
 ///
 /// `relative` is a path relative to the package root.
 fn vendor_this(relative: &Path) -> bool {
-    match relative.to_str() {
-        // Skip git config files as they're not relevant to builds most of
-        // the time and if we respect them (e.g.  in git) then it'll
-        // probably mess with the checksums when a vendor dir is checked
-        // into someone else's source control
-        Some(".gitattributes" | ".gitignore" | ".git") => false,
-
-        // Temporary Cargo files
-        Some(".cargo-ok") => false,
-
-        _ => true,
+    for component in relative.components() {
```

Comment:
Any specific reason not using [`Path::file_name`](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.file_name)?

I guess except `.git`, everthing should be a file. Not sure if it worth having a separate logic for `.git` dir.